app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -1044,14 +1026,10 @@ export class FilterChangesList extends React.Component<
     this.setState({ filterText: text })
   }
 
-  private onFileSelectionChanged = (item: IChangesListItem | null) => {
-    const rows = item
-      ? [
-          this.props.workingDirectory.files.findIndex(
-            f => f.id === item.change.id
-          ),
-        ]
-      : []
+  private onFileSelectionChanged = (items: ReadonlyArray<IChangesListItem>) => {
+    const rows = items.map(i =>
+      this.props.workingDirectory.files.findIndex(f => f.id === i.change.id)
```

Comment:
This can get expensive fast if there's lots of selected changes and lots of files. Suggest we use the map to lookup here instead

```suggestion
      this.props.workingDirectory.findFileIndexByID(i.change.id)
```